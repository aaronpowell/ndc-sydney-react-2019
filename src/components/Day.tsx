import React from "react";
import { IAgendaItem, groupAgendaTimeslots } from "../agendaUtils";
import TimeSlot from "./Timeslot";
import styles from "styled-components";

const DayContainer = styles.div`
  border: 1px solid #cdcdcd;
  margin: 40px;
  border-radius: 10px 10px 0px 0px;
`;

const Title = styles.h2`
  text-transform: capitalize;
`;

interface IDayProps {
  day: string;
  agenda: IAgendaItem[];
}

const Day: React.FC<IDayProps> = ({ day, agenda }) => {
  const timeslots = groupAgendaTimeslots(agenda);

  return (
    <DayContainer>
      <Title>{day}</Title>
      {Object.keys(timeslots).map(key => (
        <TimeSlot key={key} timeslot={key} talks={timeslots[key]} />
      ))}
    </DayContainer>
  );
};

export default Day;
