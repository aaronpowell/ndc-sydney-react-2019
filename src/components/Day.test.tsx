import React from "react";
import Day from "./Day";
import { render } from "@testing-library/react";

it("will group by times", () => {
  const agenda = [
    {
      day: "test",
      startTime: { hour: 9, minutes: 0 },
      endTime: { hour: 10, minutes: 0 },
      title: "title 1",
      speaker: "speaker 1",
      room: "test room"
    },
    {
      day: "test",
      startTime: { hour: 9, minutes: 0 },
      endTime: { hour: 10, minutes: 0 },
      title: "title 2",
      speaker: "speaker 2",
      room: "test room"
    },
    {
      day: "test",
      startTime: { hour: 10, minutes: 20 },
      endTime: { hour: 11, minutes: 20 },
      title: "title 3",
      speaker: "speaker 3",
      room: "test room"
    }
  ];

  const { asFragment } = render(<Day day="test" agenda={agenda} />);

  expect(asFragment()).toMatchSnapshot();
});
