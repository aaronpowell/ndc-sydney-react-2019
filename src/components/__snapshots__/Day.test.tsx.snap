// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`will group by times 1`] = `
<DocumentFragment>
  <div
    class="sc-bdVaJa dLXKxI"
  >
    <h2
      class="sc-bwzfXH lbWQEJ"
    >
      test
    </h2>
    <div
      class="Timeslot"
    >
      <h4>
        09:00 - 10:00
      </h4>
      <p>
        title 1 - speaker 1
      </p>
      <p>
        title 2 - speaker 2
      </p>
    </div>
    <div
      class="Timeslot"
    >
      <h4>
        10:20 - 11:20
      </h4>
      <p>
        title 3 - speaker 3
      </p>
    </div>
  </div>
</DocumentFragment>
`;
